cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -370,6 +372,22 @@ void addApplication(Id.Application id,
   WorkflowToken getWorkflowToken(Id.Workflow workflowId, String workflowRunId);
 
   /**
+   * Add node state for the given {@link Workflow} run. This method is used to update the
+   * state of the custom actions started by Workflow.
```

Comment:
Let's keep the current signature for now. Need to spend some time on the Store interface later.